src/OpenLoco/src/World/TownManager.cpp

```diff
@@ -29,12 +34,140 @@ namespace OpenLoco::TownManager
         return regs.eax;
     }
 
+    // 0x00497D70
+    static uint32_t copyTownNameToBuffer(const TownNamesObject* namesObj, uint32_t offset, uint16_t index, char* buffer)
+    {
+        auto* offsetPtr = reinterpret_cast<const std::byte*>(namesObj) + offset;
+        auto srcOffset = *reinterpret_cast<const int16_t*>(offsetPtr + index * 2);
```

Comment:
Add a comment that the first section of the town names is a table to access the individual name.

```
TownObject (HEADER) 0x1A
[
   The following is a jagged array as strings have variable length

   uint16_t stringOffsets[N]; Will offset into the following table (unusre if the following table will actually be immediately here in memory or after all 6 of the jagged array stringOffsets tables)
   [
   char string[L]; Where L represents the length of the string + null terminator
   uint8_t flags;
   ] Can have multiples of these
] Can have up to 6 of these